astropy/wcs/utils.py

```diff
@@ -632,3 +632,203 @@ def pixel_to_skycoord(xp, yp, wcs, origin=0, mode='all', cls=None):
     coords = cls(frame.realize_frame(data))
 
     return coords
+
+
+def _linear_transformation_fit(params, lon, lat, x, y, w_obj):
+    """ Objective function for fitting linear terms."""
+    pc = params[0:4]
+    crpix = params[4:6]
+
+    w_obj.wcs.pc = ((pc[0],pc[1]),(pc[2],pc[3]))
+    w_obj.wcs.crpix = crpix
+    lon2, lat2 = w_obj.wcs_pix2world(x,y,1)
+    resids = np.concatenate((lon-lon2, lat-lat2))
```

Comment:
This check can give bad results if the WCS is near the 360->0 degree crossover, suggest adding something like:
```
resids[resids > 180] = 360 - resids[resids > 180]
resids[resids < -180] = 360 + resids[resids < -180]
```

To avoid getting large residuals between eg. 359.8 deg and 0.08 deg.